### PR TITLE
Per-stream Gaussian-on-residuals fusion: SAFA + normalized landmark

### PR DESCRIPTION
**Pull request by efahnestock**
## Summary

Adds a new histogram-filter aggregator (`SafaPlusNormalizedLandmarkAggregator`) that emits a true per-patch `log p(z | patch_j)` for every patch — no row-level softmax — by summing two independent Gaussian-on-residuals log-likelihoods.

- **Image side**: SAFA-style Gaussian on the raw residual `r = sim_max − sim`. σ = 0.187.
- **Landmark side**: divide each row by `row_max`, then Gaussian on `r_norm = 1 − sim/row_max`. σ = 0.737. Equivalent to a per-row σ that scales with `row_max`, which makes a single calibrated `σ_lm` transfer across cities whose absolute landmark-similarity scales differ wildly.
- **All-zero / constant landmark rows** fall through to image-only (the `/row_max` normalization is undefined and landmark is genuinely uninformative).
- **Fusion**: sum the two per-patch log-likelihoods (conditional independence given patch). No α / softmax.

## Calibration

Both σ values calibrated on Seattle only.

### σ_img = 0.187

Half-normal MLE on per-pair image residuals from the Seattle SAFA matrix (`safa_dinov3_wag_chicago.pt`). Computed via `scripts/calibrate_sigma.py`. Same value already in production for the single-matrix aggregator.

### σ_lm = 0.737

Half-normal MLE on per-pair `r_norm` from the v6 Hungarian landmark matrix on Seattle. Cross-city consistency check (Chicago / Seattle / NewYork / post_hurricane_ian) confirms σ_MLE max/min spread is 1.12× — well below the 1.25× transferability target. Computed via `scripts/landmark_residual_exploration.py`. The script also produces 5 figures (row-max distribution, per-pair true sim overlay, normalized-residual overlay, per-city true vs negative residuals, σ bar chart) that ground the choice.

## Results

Production eval at `/data/overhead_matching/evaluation/results/260501_v6_safa_plus_norm_lm/` (full README there). Apples-to-apples comparison vs the v6 SAFA-only baseline (`260428_v6_safa_only_noise014`) on the same paths (`mappilary_equal_length/3k/<city>.json` for mappilary cities, `<city>_5k_5km_goal_directed.json` for Seattle/NY) at noise 0.141 m/√m, radii 25/50/100m:

| city | c100 (new / safa) | final err (new / safa) |
|---|---|---|
| Boston | **1680 / 3005** | **303 m / 1899 m** |
| Framingham | **2130 / 2994** | **629 m / 3001 m** |
| Gap | **649 / 689** | 10.3 m / 10.4 m |
| Middletown | **581 / 749** | 22.0 m / 18.9 m |
| NewYork | **287 / 754** | **8.83 m / 9.16 m** |
| Norway | **1076 / 1253** | **475 m / 1264 m** |
| post_hurricane_ian | **656 / 1210** | **19.2 m / 209 m** |
| SanFrancisco_mapillary | **91 / 415** | 7.4 m / 7.7 m |
| Seattle | **268 / 372** | 7.04 m / 6.66 m |
| nightdrive | **1865 / 3000** | **540 m / 1483 m** |

New aggregator wins **c100 on every one of 10 cities** and **final-position-error on 7/10**. Margins are largest on catastrophic-shift cities (Boston, Framingham, Norway, nightdrive, post_hurricane_ian) where SAFA-only confidently mis-localizes.

## Test plan

- [x] `bazel test //experimental/overhead_matching/swag/filter:adaptive_aggregators_test` — unit tests for the aggregator
  - all-zero landmark row falls through to image-only
  - constant non-zero landmark row falls through to image-only
  - patch at `row_max` gets `log_p_lm = -log(σ_lm √2π)` (zero residual)
  - residual decreases log_p_lm quadratically as `−0.5 (r_norm/σ_lm)²`
- [x] Production eval on 10 cities (this PR's headline numbers)
- [ ] Reproduce a single city via `bazel run //experimental/overhead_matching/swag/scripts:evaluate_histogram_on_paths -- --aggregator-config <city>/safa_plus_norm_lm/aggregator_config.yaml ...` (see the eval-results README for the full per-city CLI)

## Files

- `experimental/overhead_matching/swag/filter/adaptive_aggregators.py` — adds `SafaPlusNormalizedLandmarkAggregator` + config + factory branch.
- `experimental/overhead_matching/swag/filter/adaptive_aggregators_test.py` — unit tests.
- `experimental/overhead_matching/swag/scripts/calibrate_sigma.py` — σ_img derivation.
- `experimental/overhead_matching/swag/scripts/landmark_residual_exploration.py` — σ_lm derivation + cross-city consistency plots.
- `experimental/overhead_matching/swag/filter/BUILD` — adds `vigor_dataset` dep on `adaptive_aggregators` (it's been imported all along but not declared) + the test target.
- `experimental/overhead_matching/swag/scripts/BUILD` — adds the two py_binary targets.

The eval rollout (`scripts/evaluate_histogram_on_paths.py`) consumes the new aggregator with no code changes — it dispatches via the existing `AggregatorConfig` msgspec union.

## Companion archive PR

A larger workspace-state archive PR with σ-calibration tooling exploration, a differentiable filter, and a learned per-step σ/α policy lives at #625 — that one is for record only and should not be merged. The current PR is the production-ready subset.

### CODE REVIEW

File: experimental/overhead_matching/swag/data/vigor_dataset.py [hunk 1]
```diff
@@ -133,9 +132,14 @@ def series_to_dict_with_index(series: pd.Series, index_key: str = "index"):
...
 
 
+_IMAGE_SUFFIXES = {".jpg", ".jpeg", ".png"}
+
+
 def load_satellite_metadata(path: Path, zoom_level: int):
     out = []
     for p in sorted(list(path.iterdir())):
+        if p.suffix.lower() not in _IMAGE_SUFFIXES:
+            continue  # skip cache/.pt sidecars that share filename stems
```
Comment (efahnestock) [inline experimental/overhead_matching/swag/data/vigor_dataset.py]: Add a warning printout here, this shouldn't happen

File: experimental/overhead_matching/swag/data/vigor_dataset.py [hunk 2]
```diff
@@ -147,6 +151,8 @@ def load_satellite_metadata(path: Path, zoom_level: int):
 def load_panorama_metadata(path: Path, zoom_level: int):
     out = []
     for p in sorted(list(path.iterdir())):
+        if p.suffix.lower() not in _IMAGE_SUFFIXES:
+            continue  # skip cache/.pt sidecars that share filename stems
```
Comment (efahnestock) [inline experimental/overhead_matching/swag/data/vigor_dataset.py]: same here

File: experimental/overhead_matching/swag/filter/adaptive_aggregators.py [hunk 3]
```diff
@@ -36,18 +36,56 @@ class ImageLandmarkPrivilegedInformationFusionConfig(
 
 
 class EntropyAdaptiveAggregatorConfig(msgspec.Struct, **MSGSPEC_STRUCT_OPTS):
-    """Config for entropy-adaptive weighted fusion aggregator."""
+    """Config for entropy-adaptive weighted fusion aggregator.
```
Comment (efahnestock) [inline experimental/overhead_matching/swag/filter/adaptive_aggregators.py]: I don't think we use this class, so please reverse it to what it was before

File: experimental/overhead_matching/swag/filter/adaptive_aggregators.py [hunk 4]
```diff
@@ -114,6 +152,15 @@ def __init__(
     def __call__(self, pano_id: str) -> torch.Tensor:
         pano_index = self._pano_id_index.get_loc(pano_id)
```
Comment (efahnestock) [inline experimental/overhead_matching/swag/filter/adaptive_aggregators.py:151]: does this throw if not found?

File: experimental/overhead_matching/swag/filter/adaptive_aggregators.py [hunk 5]
```diff
@@ -204,19 +251,25 @@ class EntropyAdaptiveAggregator(ObservationLogLikelihoodAggregator):
 
     Computes a per-source peak sharpness confidence score (max - mean of log-probs)
     and uses it to blend the two similarity vectors before converting to log-likelihoods.
+
+    Image and landmark similarities are normalized through `log_softmax(x / sigma)`
+    with their own per-source sigmas (`image_sigma`, `landmark_sigma`) — these
+    parameterize how peaky each source's posterior is before fusion.
```
Comment (efahnestock) [inline experimental/overhead_matching/swag/filter/adaptive_aggregators.py]: again, backout changes to this aggregator

File: experimental/overhead_matching/swag/filter/adaptive_aggregators.py [hunk 6]
```diff
@@ -271,15 +324,118 @@ def from_config(
...
+        lm_sim = self.landmark_similarity_matrix[pano_index].to(self.device)
+
+        # Image-side Gaussian-on-residuals (SAFA form).
+        log_p_img = wag_observation_log_likelihood_from_similarity_matrix(
+            img_sim, self.image_sigma
+        )
+
+        # Landmark-side: per-row /max normalization, then Gaussian-on-r_norm.
+        # Fall through to image-only when the landmark row is uninformative.
+        lm_finite_mask = torch.isfinite(lm_sim)
```
Comment (efahnestock) [inline experimental/overhead_matching/swag/filter/adaptive_aggregators.py:386]: throw here instead if any values are nan/inf, don't replace with 0

File: experimental/overhead_matching/swag/filter/adaptive_aggregators.py [hunk 7]
```diff
@@ -271,15 +324,118 @@ def from_config(
...
+        if not lm_finite_mask.any():
+            return _replace_nan_with_zero(log_p_img)
```
Comment (efahnestock) [inline experimental/overhead_matching/swag/filter/adaptive_aggregators.py]: same here, throw if this becomes nan, it should be impossible

File: experimental/overhead_matching/swag/filter/adaptive_aggregators.py [hunk 8]
```diff
@@ -271,15 +324,118 @@ def from_config(
...
+        norm_sim = lm_sim / sim_max_lm
+        r_norm = 1.0 - norm_sim
+        sigma_lm = self.landmark_sigma
+        log_norm_const = -torch.log(
+            torch.sqrt(torch.tensor(2 * torch.pi, device=lm_sim.device))
+        ) - torch.log(torch.tensor(sigma_lm, device=lm_sim.device))
+        log_p_lm = log_norm_const - 0.5 * torch.square(r_norm / sigma_lm)
+
+        # If a single entry is non-finite (e.g., -inf landmark), pass through
+        # image-only at that entry to avoid NaN propagation downstream.
```
Comment (efahnestock) [inline experimental/overhead_matching/swag/filter/adaptive_aggregators.py]: again, should throw, not swap for zeros

File: experimental/overhead_matching/swag/scripts/BUILD [hunk 9]
```diff
@@ -1015,3 +1015,41 @@ py_binary(
   ]
 )
 
+py_binary(
+  name="compare_simple_v1_v3_vs_v6",
```
Comment (efahnestock) [inline experimental/overhead_matching/swag/scripts/BUILD]: remove this script/binary

File: experimental/overhead_matching/swag/scripts/BUILD [hunk 10]
```diff
@@ -1015,3 +1015,41 @@ py_binary(
...
+    requirement("numpy"),
+    requirement("matplotlib"),
+    requirement("scipy"),
+    "//common/torch:load_torch_deps",
+    "//experimental/overhead_matching/swag/data:vigor_dataset",
+    "//experimental/overhead_matching/swag/filter:adaptive_aggregators",
+  ]
+)
+
+py_binary(
```
Comment (efahnestock) [inline experimental/overhead_matching/swag/scripts/BUILD]: remove this one as well

File: experimental/overhead_matching/swag/scripts/step_through_histogram.py [hunk 11]
```diff
@@ -376,10 +378,17 @@ def main():
 
     simple_mode = args.simple
     lightweight_mode = args.lightweight
-    noise_percent = args.noise_percent if args.noise_percent is not None else eval_args.get("noise_percent", 0.02)
+    # Try the new key first, then fall back to the legacy "noise_percent" key
```
Comment (efahnestock) [inline experimental/overhead_matching/swag/scripts/step_through_histogram.py]: don't fall back, just throw

File: experimental/overhead_matching/swag/scripts/calibrate_sigma.py [hunk 12]
```diff
@@ -0,0 +1,673 @@
+"""Calibrate the observation-likelihood sigma from a precomputed similarity matrix.
```
Comment (efahnestock) [inline experimental/overhead_matching/swag/scripts/calibrate_sigma.py]: since we only use the MLE part of this, please drop the NLL things that were not used to calibrate the two sigmas on seattle
